### ISSUE TITLE
Make autoprune service to monitor diskspace and prune dataitems when low.

### ISSUE THREAD

**Issue by MRIIOT** [Ladder99/ladder99#211]
I thought retention is already set on db.
What will autoprune do?

**Comment by bburns** [comment]
retention works, but it's guesswork as to how long you could keep data around - 1w? 1y?

so could have a service prune the db when space gets low. 

it would avoid clients using ladder99 and crashing their disks.

not sure how other similar programs handle the issue.

**Comment by tukusejssirs** [comment]
@bburns, should this issue be labelled with `component:autoprune`?

**Comment by MRIIOT** [comment]
@tukusejssirs I don't think this is a bug.  This is a new feature.  The speed of which disk is consumed is all dependent on disk size.  Current retention is guesswork, but it works when the guess is correct.

**Comment by tukusejssirs** [comment]
> I don't think this is a bug. This is a new feature. The speed of which disk is consumed is all dependent on disk size. Current retention is guesswork, but it works when the guess is correct.

Well, it depends on the point of view. While I understand your point, IMHO if there is at least one case when the current retention fails (when the guess is incorrect), it should be considered a bug. Yes, this bug is not of a top priority and yes, it does not happen a lot (only rarely). Although it is a bug, as it is of low priority, it does not need to be fixed right away.

YMMV. If you still think it is not a bug, okay, we can change the label to `feature`.
